fix(cos-462, cos-508): Preview modal + ui improvements

Kasia9090 opened the pull request:
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
### Summary by CodeRabbit

**New Features:**
- Introduced `IssuePreviewModal` for detailed issue previews, enhancing user understanding of issue status and details.
- Added `IssueCommentCard` to display issue comments, improving user engagement and communication.
- Implemented `PriorityBadge` to visually represent issue priority levels, aiding in issue triage and management.
- Replaced specific `ViewInSlackButton` and `ViewInIntercomButton` with a generic `ViewInExternalAppButton`, increasing flexibility and consistency across different platforms.

**Improvements:**
- Enhanced `IssueCard` and `IssueStub` components for better issue status representation and interaction.
- Expanded GraphQL queries to fetch more detailed issue and interaction event data, providing a more comprehensive view of timeline events.

**Style:**
- Adjusted styles and layouts for better user interface experience and consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Kasia9090 committed:
WIP: "Refactor ViewInExternalAppButton, removing duplication

Created a new ViewInExternalAppButton component that abstracts the shared functionality of ViewInIntercomButton and ViewInSlackButton. This leads to code that is easier to maintain and less prone to bugs. Also updated the places where these old components were used with the new ViewInExternalAppButton component."

Kasia9090 committed:
Add IssuePreviewModal and restructure timeline issue card

Made major enhancements to the issue card functionality in the timeline view. Apart from various refinements to the UI and layout, significant changes include:

- Added a new AlertSquare SVG icon in the media folder
- Created a new dynamic PriorityBadge component to display issue priority visually
- Built the IssuePreviewModal to show issue details upon clicking the issue card. This involved a major restructure of existing code in IssuePreviewModal.tsx and improvement of related functionalities. Call to this is currently mocked.
- Added mock data in IssueStub for testing
- Integrated the IssuePreviewModal in the TimelineEventPreviewModal
- Included adjustment in styles in IssueCommentCard for better visual aesthetics

Kasia9090 committed:
(fix|COS-508): Issues ui polish

acalinica commented on packages/apps/spaces/app/organization/[id]/src/components/Timeline/events/issue/IssueStub.tsx:
```diff
@@ -44,24 +28,17 @@ export const IssueStub: FC<{ data: IssueWithAliases }> = ({ data }) => {
       position='unset'
       maxW={476}
-      cursor='default' // todo change to pointer when modal is ready
+      cursor='cursor'
```
He might be right :))

Kasia9090 commented on packages/apps/spaces/app/organization/[id]/src/components/Timeline/events/issue/IssueStub.tsx:
```diff
@@ -7,8 +7,7 @@ import { IssueBgPattern } from '@ui/media/logos/IssueBgPattern';
 import { Tag, TagLabel } from '@ui/presentation/Tag';
 import { CustomTicketTearStyle } from './styles';
 import { IssueWithAliases } from '@organization/src/components/Timeline/types';
-import { getExternalUrl } from '@spaces/utils/getExternalLink';
-import { toastError } from '@ui/presentation/Toast';
+import { useTimelineEventPreviewContext } from '@organization/src/components/Timeline/preview/context/TimelineEventPreviewContext';
 function getStatusColor(status: string) {
   if (status === 'solved' || status === 'closed') {
     return 'gray';
```
it's a binary check, gray or blue. I'm not convinced mapping object will improve anything here

Kasia9090 committed:
Added Zendesk logo, adjusted styles, and expanded issue data

Introduced Zendesk's logo for use in the ViewInExternalAppButton component. Several styles were adjusted, including zIndex in the IssueStub style and cursor behaviour in IssueCommentCard.
